### ISSUE TITLE
Does this project support urllib3 directly?

### ISSUE THREAD

**Issue by FredHappyface** [requests-cache/requests-cache#981]
## Background 

I've been investigating persistent cache options for urllib3 and cake across the following issue https://github.com/urllib3/urllib3/issues/2166

## The Question 

So I'm wondering if this lib supports urllib3 directly and if not would you accept a pr to add this functionality? Currently I'm not sure on how this would be implemented but can have a play around

## Other options 

I *could* make some urllib3-cache library however this would result in a bunch of duplicated work. I'd rather spend that time contributing to an existing lib 

Thanks for your time and an awesome project :)

**Comment by JWCook** [comment]
No, requests-cache doesn't support urllib3 directly. I have tried to keep some of the library's modules HTTP client-agnostic, mainly the cache policy and serialization. Some of the other modules like request matching (cache keys) could be made client-agnostic with a few tweaks. Also, [this module](https://github.com/requests-cache/requests-cache/blob/main/requests_cache/models/raw_response.py) emulates urllib3's HTTPResponse class, which `requests` uses internally.

However, most of this library's public API is very specific to `requests`, particularly its Session, Request, and Response interfaces.

I've considered the possibility of some shared utilities that could work with multiple HTTP clients, but my sense so far is that trying to make that work would be messier than just duplicating code. For aiohttp, for example, I opted to make a [completely separate library](https://github.com/requests-cache/aiohttp-client-cache), although the main factor there was the difficulty of reusing sync and async logic, which wouldn't apply to urllib3.

While I'm probably not interested in a massive refactor of requests-cache to add multi-client support, I may be open to making some smaller incremental changes to make it easier for you to add urllib3 support in either a fork or a separate library that depends on requests-cache.
